src/ray/gcs/gcs_server/gcs_autoscaler_state_manager.cc

```diff
@@ -276,7 +280,34 @@ void GcsAutoscalerStateManager::GetNodeStates(
 void GcsAutoscalerStateManager::HandleDrainNode(
     rpc::autoscaler::DrainNodeRequest request,
     rpc::autoscaler::DrainNodeReply *reply,
-    rpc::SendReplyCallback send_reply_callback) {}
+    rpc::SendReplyCallback send_reply_callback) {
+  const NodeID node_id = NodeID::FromBinary(request.node_id());
+  RAY_LOG(INFO) << "HandleDrainNode " << node_id.Hex()
+                << ", reason: " << request.reason_message();
+
+  auto node = gcs_node_manager_.GetAliveNode(node_id);
+  if (!node.has_value()) {
+    // The node is dead so treat it as drained.
+    reply->set_is_accepted(true);
```

Comment:
should we differentiate this case? would we ever run into race condition?

Reply:
This is possible: the node is crashed when autoscaler tries to drain it.